content/articles/2017-12-29-dbplyr-1-2.Rmd

```diff
@@ -0,0 +1,129 @@
+
+- `copy_to()` now allows to take the results from one database and cache them in a different database.
+
+- stringr functions: `str_length()`, `str_to_upper()`, `str_to_lower()`, `str_replace_all()`, `str_detect()`, and  `str_trim()` are now supported.
+
+- `in_schema()` should now work in more places, particularly in `copy_to()` 
+
+- Add support for temporary tables in Microsoft SQL Server.  Additionally, the issue of certain operators working only in `mutate()` or only in `filter()` has been resolved.
+
+-  New for developers: `remote_name()`, `remote_con()`, `remote_src()`, `remote_query()` and  `remote_query_plan()` provide a standard API for get metadata about a remote `tbl`. Also new, `sql_aggregate()` and `win_aggregate()` for generating SQL and windowed SQL functions for aggregates. 
+
+## Cache query results
+
+The `copy_to()` function is now able to **cache the results of a set of dplyr database transformations**.  An additional advantage is that `copy_to()` uses the database to run and cache the data, so there is no data being transmitted back to R. For this feature to work, the user needs sufficient rights to create temporary tables in the database.  
+
+```{r}
+con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
+copy_to(con, rownames_to_column(mtcars), "mtcars")
+
+only_auto <- tbl(con, "mtcars") %>%
+  filter(am == 1) %>%
+  select(mpg, cyl, am)
+copy_to(con, only_auto, "auto")                     # New in dbplyr 1.2.0
+tbl(con, "auto") %>%
+  head()
+
+```
+
+## Cache in different database 
+
+If the set of dplyr transformations need to be cached to a different database, `copy_to()` is smart enough to download the data into memory in R, and then uploads the data to the target database. 
+
+```{r}
+second_con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
+copy_to(second_con, only_auto, "auto")               # New in dbplyr 1.2.0
+tbl(second_con, "auto") %>%
+  head()
+```
+
+## stringr functions
+
+Six stringr functions are now supported.  Regular expression support varies from database to database, but most simple regular expressions should be ok.  The functions are:
+
+1. `str_length()` 
+2. `str_to_upper()`
+3. `str_to_lower()` 
+4. `str_replace_all()`
+5. `str_detect()`
+6. `str_trim()` 
+
+```{r}
+tbl(con, "mtcars") %>% 
+  head() %>%
+  select(rowname) %>%
+  mutate(s1 = str_length(rowname),                   # New in dbplyr 1.2.0
+         s2 = str_to_upper(rowname),                 # New in dbplyr 1.2.0
+         s3 = str_to_lower(rowname),                 # New in dbplyr 1.2.0
+         s4 = str_replace_all(rowname, "M", "X"),    # New in dbplyr 1.2.0
+         s5 = str_detect(rowname, "M"),              # New in dbplyr 1.2.0
+         s6 = str_trim(rowname))                     # New in dbplyr 1.2.0
```

Comment:
Can you please follow tidyverse indenting style here? (You can now use the styler package to correct automatically).

I think you can delete the comments here